tools/clientSimulator/README.md

```diff
@@ -0,0 +1,241 @@
+DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# Client simulator
+
+## Project
+
+This project allows users to test Parameter Framework configurations.  It
+allows writing scenario, mixing criterion changes and arbitrary script
+execution.  The goal of such tests are to recreate, under a test harness, the
+real use-cases that will be executed by a Parameter Framework.
+
+In short: it eases the development of automated Parameter Framework
+configuration functional tests.
+
+## How to run tests
+
+You can run tests using pfClientSimulator.py.
+
+You have to run the script with at least the test directory argument:
+
+* `test_directory`: path to the test directory
+
+Different options are available:
+
+* `-s`, `--scenario`    : precise the scenario number to launch
+* `-i`, `--interactive` : run in interactive mode to test new vectors
+* `-v`, `--verbose`     : make the script talk on stdin
+* `-c`, `--coverage`    : generate the html coverage report at the end of the script
+* `-h`, `--help`        : show options recap
+
+To see available scenarios in the test directory, you can run:
+
+    ./pfClientSimulator.py path/to/tests/
+
+The script will wait for a choice and the desired scenario will be
+launched. You can also directly select a scenario with -`s` option.
+
+## How to create a test environment
+
+### Test Directory
+
+A test directory should contains a `conf.json` file containing:
+
+- The desired command prefix (optional; e.g. `adb shell` in order to execute
+  tests on an android board or empty to execute locally).
+- The criterion file path (see
+  [this README](https://github.com/01org/parameter-framework/tree/master/tools/xmlGenerator#domaingeneratorpy)).
+- The absolute path to the Parameter Framework toplevel configuration file.
+- The path to the directory containing the scenario files.
+- The path to the scripts definitions file (optional) (see below).
+- The path to the actions definitions (aka "ActionGatherer") file (optional)
+  (see below).
+- The absolute path to the log output file (optional).
+- A setup script (inline shell) (optional).
+- The absolute paths to test-platform and remote-process executables (or the
+  executables' names if they are in the PATH).
+- The host and port on which the test-platform and the Parameter Framework
+  instance are listening for commands.
+- The absolute path to the directory containing the coverage generation tool
+  (optional; for coverage only).
```

Comment:
https://github.com/dawagner/parameter-framework/tree/master/tools/coverage

I'll reference it in the readme.